File: app/src/main/java/paul/barthuel/go4lunch/data/retrofit/AutocompleteRepository.java
package paul.barthuel.go4lunch.data.retrofit;

import android.location.Location;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import paul.barthuel.go4lunch.data.model.autocomplet.Autocomplete;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AutocompleteRepository {

    public LiveData<Autocomplete> getAutocompleteForLocation(@NonNull String userInput, @NonNull Location location) {
        final MutableLiveData<Autocomplete> liveData = new MutableLiveData<>();
        RetrofitService.getInstance().getGooglePlacesAPI().getAutocompleteSearch(userInput,
                location.getLatitude() + "," + location.getLongitude(),
                500,
                "AIzaSyDf9lQFMPnggxP8jYVT8NvGxmSQjuhNrNs")
                .enqueue(new Callback<Autocomplete>() {
                    @Override
                    public void onResponse(@NonNull Call<Autocomplete> call, @NonNull Response<Autocomplete> response) {
                        if (response.body() != null) {
                            liveData.postValue(response.body());
                        }
                    }

                    @Override
                    public void onFailure(@NonNull Call<Autocomplete> call, @NonNull Throwable t) {
                        t.printStackTrace();
                    }
                });

        return liveData;
    }
}

File: app/src/main/java/paul/barthuel/go4lunch/data/retrofit/AutocompleteRepository.kt
package paul.barthuel.go4lunch.data.retrofit

import android.location.Location
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import paul.barthuel.go4lunch.data.model.autocomplet.Autocomplete
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class AutocompleteRepository {
    fun getAutocompleteForLocation(userInput: String, location: Location): LiveData<Autocomplete> {
        val liveData = MutableLiveData<Autocomplete>()
        RetrofitService.instance?.googlePlacesAPI?.getAutocompleteSearch(userInput,
                location.latitude.toString() + "," + location.longitude,
                500,
                "AIzaSyDf9lQFMPnggxP8jYVT8NvGxmSQjuhNrNs")
                ?.enqueue(object : Callback<Autocomplete?> {
                    override fun onResponse(call: Call<Autocomplete?>, response: Response<Autocomplete?>) {
                        if (response.body() != null) {
                            liveData.postValue(response.body())
                        }
                    }

                    override fun onFailure(call: Call<Autocomplete?>, t: Throwable) {
                        t.printStackTrace()
                    }
                })
        return liveData
    }
}